tests/test_recipe.py

```diff
@@ -115,13 +115,23 @@ def test_download_url_not_set(self):
             mock.call('Skipping test_recipe download as no URL is set')]
 
     @staticmethod
-    def get_dummy_python_recipe_for_download_tests():
+    def get_dummy_python_recipe_for_download_tests(good_digests=[], bad_digests=[]):
```

Comment:
I'd rather avoid mutable default arguments.
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments
I know it doesn't look as elegant, but to avoid potential debugging headaches I'd recommend doing this instead:
```python
def get_dummy_python_recipe_for_download_tests(good_digests=None, bad_digests=None):
    good_digests = good_digests or []
    bad_digests = bad_digests or []
```